lib/Eclair/LLVM/BTree.hs

```diff
@@ -170,18 +167,17 @@ generateTypes sizes = mdo
   pure $ Types
     { btreeTy = btreeTy
     , iteratorTy = btreeIteratorTy
-    , nodeSizeTy = nodeSizeTy
-    , nodeTypeTy = nodeTypeTy
+    , nodeSizeTy = nodeSizeTy'
+    , nodeTypeTy = nodeTypeTy'
     , nodeTy = nodeTy
```

Comment:
Interesting that it doesn't warn on these recursively defined types. But it's fine, leave as-is.